packages/angular/build/src/builders/extract-i18n/builder.ts

```diff
@@ -32,6 +32,17 @@ export async function execute(
     return { success: false };
   }
 
+  const { projectType } = (await context.getProjectMetadata(projectName)) as {
+    projectType?: string;
+  };
+  if (projectType !== 'application') {
+    context.logger.error(
+      `The 'extract-i18n' builder requires the project type to be an application.`,
```

Comment:
I think a more descriptive error message would be useful to make sure we're clear about what project we're referring to and what we mean by "project type" in this context. How about something like:

> Tried to extract from `${projectName}` with `projectType` `${projectType}`, which is not supported. The `extract-i18n` builder can only extract from applications.

This way we're calling out the affected project (there could be multiple) and referencing the relevant `projectType` property (which is going to print `library` in this case).